### PR TITLE
mcs: support rules http interface in scheduling server

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/5839

It can be reviewed after #7184 merged and #7200 solved

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by lhy1024** [comment]
ptal @rleungx @jmpotato @cabinfeverb

**Review by rleungx — APPROVED** [review]

File: server/api/rule.go [hunk 1]
```diff
@@ -180,20 +193,25 @@ func (h *ruleHandler) preCheckForRegionAndRule(w http.ResponseWriter, r *http.Re
...
+	}
+	if err != nil {
+		h.rd.JSON(w, http.StatusInternalServerError, err.Error())
 		return
 	}
 	keyHex := mux.Vars(r)["key"]
 	key, err := hex.DecodeString(keyHex)
 	if err != nil {
-		h.rd.JSON(w, http.StatusBadRequest, "key should be in hex format")
+		h.rd.JSON(w, http.StatusBadRequest, errs.ErrKeyFormat.Error())
```
Comment (CabinfeverB) [inline server/api/rule.go:211]: Do we need to wrap `err` in `ErrKeyFormat`？

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**